atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -186,7 +188,15 @@ private void addSnapshotListener() {
     }
 
     final var snapshotChunk = new SnapshotChunkImpl();
-    snapshotChunk.wrap(new UnsafeBuffer(request.data()), 0, request.data().capacity());
+    final var snapshotChunkBuffer = new UnsafeBuffer(request.data());
+    if (!snapshotChunk.tryWrap(snapshotChunkBuffer)) {
```

Comment:
The name `tryWrap` is a misnomer. It only checks the headers, but it doesn't wrap the content. So we have to call `snapshotChunk.wrap()` again.

Reply:
I see two options: one is we wrap in `tryWrap` (assuming the header is correct), the other is we rename it to `canParse` or some such (though it's also a bit misleading since we do wrap the header decoder, which means we do mutate the state). So I'd lean for the former

Reply:
Isn't tryWrap from generated code?

Reply:
I think it is not generated.